### ISSUE TITLE
buildsystem doesn't honour distro-wide compiler flags

### ISSUE THREAD

**Issue by sharkcz** [opencryptoki/libica#14]
Probably after commit 81971f01 the buildsystem stopped to honour distro-wide compiler flags that are passed via CFLAGS/CXXFLAGS when calling the configure script. Because distro linker flags are passed one ends with eg.
````
gcc -DPACKAGE_NAME=\"libica\" -DPACKAGE_TARNAME=\"libica\" -DPACKAGE_VERSION=\"3.3.0\" -DPACKAGE_STRING=\"libica\ 3.3.0\" -DPACKAGE_BUGREPORT=\"steuer@linux.vnet.ibm.com\" -DPACKAGE_URL=\"https://github.com/opencryptoki/libica\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -D__EXTENSIONS__=1 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D_POSIX_PTHREAD_SEMANTICS=1 -D_TANDEM_SOURCE=1 -DHAVE_FCNTL_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STDDEF_H=1 -DHAVE_STDINT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_FILE_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_SYSLOG_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -DHAVE_MALLOC=1 -DHAVE_STDLIB_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_PARAM_H=1 -DHAVE_GETPAGESIZE=1 -DHAVE_MMAP=1 -DHAVE_BZERO=1 -DHAVE_FTRUNCATE=1 -DHAVE_GETTIMEOFDAY=1 -DHAVE_MEMCHR=1 -DHAVE_MEMSET=1 -DHAVE_MUNMAP=1 -DHAVE_STRCASECMP=1 -DHAVE_STRERROR=1 -DHAVE_STRSTR=1 -DHAVE_STRTOL=1 -DHAVE_SETENV=1 -DHAVE_STRTOULL=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DPACKAGE=\"libica\" -DVERSION=\"3.3.0\" -I.    -Wall -Wextra -mzarch -DICA_FIPS -O3 -D_FORTIFY_SOURCE=2 -I./include -I./../include  -MT icainfo-icainfo.o -MD -MP -MF .deps/icainfo-icainfo.Tpo -c -o icainfo-icainfo.o `test -f 'icainfo.c' || echo './'`icainfo.c
mv -f .deps/icainfo-icainfo.Tpo .deps/icainfo-icainfo.Po
/bin/sh ../libtool  --tag=CC   --mode=link gcc -Wall -Wextra -mzarch -DICA_FIPS -O3 -D_FORTIFY_SOURCE=2 -I./include -I./../include   -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -o icainfo icainfo-icainfo.o libica.la 
libtool: link: gcc -Wall -Wextra -mzarch -DICA_FIPS -O3 -D_FORTIFY_SOURCE=2 -I./include -I./../include -Wl,-z -Wl,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -o .libs/icainfo icainfo-icainfo.o  ./.libs/libica.so -lrt -lcrypto -Wl,-rpath -Wl,/usr/local/lib
/usr/bin/ld: icainfo-icainfo.o: `puts@@GLIBC_2.2' non-PLT reloc for symbol defined in shared library and accessed from executable (rebuild file with -fPIC ?)
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:517: icainfo] Error 1
````
Right now I don't see any simple workaround, but I'll look further.

**Comment by sharkcz** [comment]
Yes, it solves the problem.

**Comment by p-steuer** [comment]
Okay thanks, i made the above buildsystem patch available via a 3.3.1 bug fix release.
